magefile.go

```diff
@@ -181,3 +196,144 @@ func licenser(mode licenserMode) error {
 
 	return nil
 }
+
+type Benchmark mg.Namespace
+
+func InstallBenchStat() error {
+	err := gotool.Install(gotool.Install.Package(goBenchstats))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (Benchmark) Run(ctx context.Context) error {
+	outputFile := os.Getenv("OUTPUT")
+	mg.Deps(InstallBenchStat)
+	fmt.Println(">> go benchmark:", "Testing")
+	args := []string{
+		"test",
+		fmt.Sprintf("-count=%d", benchmarkCount),
+		"-bench=.",
+		"-run=Bench#",
+	}
+	for _, pkg := range goBenchmarkPackages {
+		args = append(args, filepath.Join("github.com/elastic/elastic-agent-shipper-client", pkg, "..."))
+	}
+	goTestBench := makeCommand(ctx, nil, "go", args...)
+
+	// Wire up the outputs.
+	var outputs []io.Writer
+	if outputFile != "" {
+		fileOutput, err := os.Create(createDir(outputFile))
+		if err != nil {
+			fmt.Errorf("failed to create go test output file: %w", err)
+		}
+		defer fileOutput.Close()
+		outputs = append(outputs, fileOutput)
+	}
+	output := io.MultiWriter(outputs...)
+	goTestBench.Stdout = io.MultiWriter(output, os.Stdout)
+	goTestBench.Stderr = io.MultiWriter(output, os.Stderr)
+
+	err := goTestBench.Run()
+
+	var goTestErr *exec.ExitError
+	if err != nil {
+		switch {
+		case errors.As(err, &goTestErr):
+			return fmt.Errorf("failed to execute go test -bench command: %w", err)
+		default:
+			return fmt.Errorf("failed to execute go test -bench command %w", err)
+		}
+	}
+
+	if goTestErr != nil {
+		// No packages were tested. Probably the code didn't compile.
+		return goTestErr
+	}
+
+	return nil
+}
+
+func (Benchmark) Diff(ctx context.Context) error {
```

Comment:
Can we add godocs to the new mage targets? The rest of them have docs, and it's nice for usability/consistency: 
```
  benchmark:diff
  benchmark:run
  check                  runs all the checks
  checkLicenseHeaders    checks ASL2 headers in .go files
  generateGo             regenerates the Go files out of .proto files
  installBenchStat
  installLicenser        installs the go-licenser.
  installProtoGo         installs required plugins for protoc
  license                applies the right license header.
  update                 updates all the generated code out of the spec
```